src/index/base.cpp

```diff
@@ -426,11 +442,8 @@ bool BaseIndex::Start()
         LOCK(m_mutex);
         m_notifications = std::move(notifications);
         m_handler = std::move(handler);
-    }
-
-    const CBlockIndex* index = m_best_block_index.load();
-    if (!CustomInit(index ? std::make_optional(interfaces::BlockKey{index->GetBlockHash(), index->nHeight}) : std::nullopt)) {
-        return false;
+        assert(m_notifications->m_init_result);
+        if (!m_notifications->m_init_result) return false;
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/24230#discussion_r820141011

> c1bdcdf3388792b7c17e576170167f0be883fa4a: why both an assert and a conditional? Or did you mean `m_init_result.value()` in the second line? Maybe also consider being explicit (`has_value()`) for the `std::optional<bool>` in the assert to avoid confusion.

Good catch. I was intending to use * operator, but I followed your suggestion instead since that seems more readable.

Note: bug here was wasn't actually present in the full PR. Bug was temporarily introduced here in commit c1bdcdf3388792b7c17e576170167f0be883fa4a, but fixed by later commit fb164ac88e8e5f7b7bb448637a06da6ae299e578